ios Counter Example fails to build

When trying to build the ios counter example, i am getting an error:
```
ContentView.swift:14:81: error: cannot convert value of type '() -> CounterRootComponent' to expected argument type '(ComponentContext) -> CounterRootComponent'
    private var componentHolder = ComponentHolder(factory: CounterRootComponent.init)
```

<img width="953" alt="Screenshot 2022-04-11 at 09 32 22" src="https://user-images.githubusercontent.com/3853006/162697090-d6851ef5-64a9-4e42-b3da-bc07c18db29c.png">

I think i remember seeing something similar before and was to do with not including the decompose lib within the generate framework - maybe a red herring though

--- a/deps.versions.toml
+++ b/deps.versions.toml
@@ -3,7 +3,7 @@
 decompose = "0.6.0"
 kotlin = "1.6.10"
 essenty = "0.2.2"
-reaktive = "1.2.0"
+reaktive = "1.2.1"
 junit = "4.13.2"
 jetbrainsCompose = "1.1.0"
 jetbrainsKotlinWrappers = "0.0.1-pre.325-kotlin-1.6.10"

--- a/sample/counter/ios-app/ios-app/ContentView.swift
+++ b/sample/counter/ios-app/ios-app/ContentView.swift
@@ -11,7 +11,13 @@ import Counter
 
 struct ContentView: View {
     @State
-    private var componentHolder = ComponentHolder(factory: CounterRootComponent.init)
+    private var componentHolder = ComponentHolder { context in
+        CounterRootComponent(
+            componentContext: context,
+            deepLink: CounterRootComponentDeepLinkNone.shared,
+            webHistoryController: nil
+        )
+    }
     
     var body: some View {
         CounterRootView(componentHolder.component)
@@ -35,7 +41,7 @@ class ComponentHolder<T> {
         let component = factory(DefaultComponentContext(lifecycle: lifecycle))
         self.lifecycle = lifecycle
         self.component = component
-
+        
         lifecycle.onCreate()
     }
     

--- a/sample/counter/shared/build.gradle.kts
+++ b/sample/counter/shared/build.gradle.kts
@@ -13,8 +13,8 @@ setupMultiplatform {
     targets(
         Target.Android,
         Target.Js(mode = Target.Js.Mode.IR),
-        Target.Ios(isAppleSiliconEnabled = false),
-        Target.WatchOs(isAppleSiliconEnabled = false),
+        Target.Ios(),
+        Target.WatchOs(),
     )
 }
 

--- a/sample/counter/watchos-app/Counter WatchKit Extension/ContentView.swift
+++ b/sample/counter/watchos-app/Counter WatchKit Extension/ContentView.swift
@@ -11,7 +11,13 @@ import Counter
 struct ContentView: View {
     
     @State
-    private var componentHolder = ComponentHolder(factory: CounterRootComponent.init)
+    private var componentHolder = ComponentHolder { context in
+        CounterRootComponent(
+            componentContext: context,
+            deepLink: CounterRootComponentDeepLinkNone.shared,
+            webHistoryController: nil
+        )
+    }
     
     var body: some View {
         CounterRootView(componentHolder.component)
@@ -35,7 +41,7 @@ class ComponentHolder<T> {
         let component = factory(DefaultComponentContext(lifecycle: lifecycle))
         self.lifecycle = lifecycle
         self.component = component
-
+        
         lifecycle.onCreate()
     }
     

--- a/sample/master-detail/shared/build.gradle.kts
+++ b/sample/master-detail/shared/build.gradle.kts
@@ -13,7 +13,7 @@ setupMultiplatform {
         Target.Android,
         Target.Jvm,
         Target.Js(mode = Target.Js.Mode.IR),
-        Target.Ios(isAppleSiliconEnabled = false),
+        Target.Ios(),
     )
 }
 

